### PR TITLE
fix(queue): get_rule_checks_status return value

### PR DESCRIPTION
**Pull request by sileht**
When some CIs are pending and some success, get_rule_checks_status may
return FAILURE instead of PENDING

This change fixes that.

Change-Id: I9acb38f149b1174196d13bb5d5fe3c18f9ed5048

### CODE REVIEW

**Comment by sileht** [comment]
This pull request is part of a stack:
1. feat(refactor): move branch protection checks earlier in the process ([#2734](https://github.com/Mergifyio/mergify-engine/pull/2734))
1. fix(queue): get_rule_checks_status return value ([#2735](https://github.com/Mergifyio/mergify-engine/pull/2735))

File: mergify_engine/context.py [hunk 1]
```diff
@@ -778,6 +778,13 @@ async def _get_consolidated_data(self, name: str) -> ContextAttributeType:
             comments, _ = await self.consolidated_reviews()
             return [r["user"]["login"] for r in comments if r["state"] == "COMMENTED"]
 
+        elif name == "check-success-or-neutral-or-pending":
```
Comment (jd) [inline mergify_engine/context.py:781]: Do we want to add this in the documentation?

Comment (sileht) [reply mergify_engine/context.py:781]: No it's internal only.

**Review by jd — APPROVED** [review]

**Comment by sileht** [comment]
@Mergifyio rebase

**Pull request merged**